### DISCUSSION TITLE
Personal Device Library Selector

### DISCUSSION THREAD

**Discussion by MaxwellAMooney** [Pirate-MIDI/Pirate-MIDI-Features-Bug-Tracking#111] [Editor App Feature Requests]
I'd like to see the ability to select from the Device Library which devices you own so that way when you want to select a device to control a parameter of in the foot switch editor, you don't have to select "Effect > Line 6 > HX Stomp" every single time you want to program a button. You can then select from only the devices you own, so you won't have to scroll the through very large Device Library every time you need to select a parameter, you can just select which device from your own Personal Library. 

So let's say you've selected HX Stomp, Boss RV-500, and Walrus D1 to be in your library. 

Now, when editing what FS 1 will do, you can select from one of those three devices in a drop down menu rather than having to select through every make and model in the entire device library for every single footswitch edit for every single bank. 

I also think it would be cool to have a "favorites" selection option that could allow you to save presets that you often use and you can drag and drop those when you want. For example, if you frequently use an expression pedal to affect the LFO speed of a trem in the HX Stomp, you could have that saved as a preset in your personal library.

**Comment by timmytimmyc** [comment]
I got the chance to get hold of a Morningstar MC6 while waiting for my Bridge6, and their User Library is great.

I hope something similar will happen in the editor, as it is a total game changer.

Tags, keyboard searches, preset MIDI channels, the ability to fetch things from the Device Library as well, the possibility to change one assignation within the user library then to mass update every preset from there.
